src/bootstrap/test.rs

```diff
@@ -1443,6 +1443,9 @@ note: if you're sure you want to do this, please open an issue as to why. In the
         if builder.config.rust_optimize_tests {
             cmd.arg("--optimize-tests");
         }
+        if builder.config.rust_randomize_layout {
+            cmd.arg("--rust-randomized-layout");
+        }
```

Comment:
I've been meaning to refactor this for a while, your change only affects compiletest :( there's no one place to put it so it also affects src/test/{codegen,run-make,debuginfo}.

compiletest seems good enough for now, but it would be nice to add this everywhere else if you have time.

Reply:
That flag is there for the `needs-deterministic-layouts` directive. As long as those test types don't get broken by randomization we won't need it.

Reply:
>  As long as those test types don't get broken by randomization we won't need it.

well, that's what I'm worried about :sweat_smile: but given this is all internal-only, I'm fine with landing this for only a subset of the testsuite.